File: app/src/main/java/com/lifedawn/bestweather/room/dto/FavoriteAddressDto.java
package com.lifedawn.bestweather.room.dto;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.io.Serializable;

@Entity(tableName = "favorite_address_table")
public class FavoriteAddressDto implements Serializable {

	@PrimaryKey(autoGenerate = true)
	@ColumnInfo(name = "id")
	private Integer id;

	@ColumnInfo
	private String displayName;

	@ColumnInfo
	private String countryName;

	@ColumnInfo
	private String countryCode;

	@ColumnInfo
	private String latitude;

	@ColumnInfo
	private String longitude;

	@ColumnInfo
	private String zoneId;

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getDisplayName() {
		return displayName;
	}

	public void setDisplayName(String displayName) {
		this.displayName = displayName;
	}

	public String getCountryName() {
		return countryName;
	}

	public void setCountryName(String countryName) {
		this.countryName = countryName;
	}

	public String getLatitude() {
		return latitude;
	}

	public void setLatitude(String latitude) {
		this.latitude = latitude;
	}

	public String getLongitude() {
		return longitude;
	}

	public void setLongitude(String longitude) {
		this.longitude = longitude;
	}

	public void setCountryCode(String countryCode) {
		this.countryCode = countryCode;
	}

	public String getCountryCode() {
		return countryCode;
	}

	public void setZoneId(String zoneId) {
		this.zoneId = zoneId;
	}

	public String getZoneId() {
		return zoneId;
	}
}

File: app/src/main/java/com/lifedawn/bestweather/data/local/room/dto/FavoriteAddressDto.kt
package com.lifedawn.bestweather.data.local.room.dto

import androidx.room.PrimaryKey
import androidx.room.ColumnInfo
import androidx.room.Entity

@Entity(tableName = "favorite_address_table")
class FavoriteAddressDto {
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "id") var id: Int? = null
    @ColumnInfo var displayName: String? = null
    @ColumnInfo var countryName: String? = null
    @ColumnInfo var countryCode: String? = null
    @ColumnInfo var latitude: String? = null
    @ColumnInfo var longitude: String? = null
    @ColumnInfo var zoneId: String? = null
}